recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,201 @@
+            self.requires("qt/5.15.2")
+
+        if self.options.vtk_smp_implementation == "tbb":
+            self.requires("TBB")
+
+        if self.options.vtk_use_mpi:
+            self.requires("openmpi")
+
+    def package(self):
+        self.copy("Copyright.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "vtk"))
+        self.copy("modules.json", dst="bin", src=self.build_folder)
+    
+    def filter_dep(self, name, dependencies, enabled_pkg):
+        ls_dep = []
+        for dep in dependencies:
+            if dep in enabled_pkg.keys() and dep != name:
+                ls_dep.append(enabled_pkg[dep]['alias'])
+            elif dep == "VTK::RenderingCore" and not "VTK::RenderingOpenGL2" in dependencies:
+                ls_dep.append("RenderingOpenGL2")
+
+        return ls_dep
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "VTK"
+        self.cpp_info.names["cmake_find_package_multi"] = "VTK"
+        lib = [i.split('-')[0] for i in tools.collect_libs(self)]
+
+        pkg_ver = [int(i) for i in self.version.split('.')]
+        base_include_dir = os.path.join('include','vtk-%d.%d' % (pkg_ver[0], pkg_ver[1]))
+
+        with open(os.path.join(self.package_folder, "bin", "modules.json")) as reader:
+            vtk_modules = json.load(reader)
+        
+        enabled_pkg = {}
+        for name, content in vtk_modules["modules"].items():
+            lib_name = content["library_name"]
+            if content["enabled"] and lib_name in lib:
+                alias = name.split("::")[1]
+                enabled_pkg[name] = {"libname": lib_name, 'alias' : alias}
+        
+        for name, info in enabled_pkg.items():
+            content = vtk_modules["modules"][name]
+            alias = info['alias']
+            lib_name = info['libname']
+
+            self.cpp_info.components[alias].names["cmake_find_package"]  = alias
+            self.cpp_info.components[alias].names["cmake_find_package_multi"] = alias
+            self.cpp_info.components[alias].libs = ['%s-%d.%d' % ( lib_name, pkg_ver[0], pkg_ver[1])]
+            dep = self.filter_dep(name, content['depends'], enabled_pkg)
+            opt_dep = self.filter_dep(name, content['private_depends'], enabled_pkg)
+            self.cpp_info.components[alias].requires.extend(dep)
+            self.cpp_info.components[alias].requires.extend(opt_dep)
+            
+            if name == "VTK::GUISupportQt" :
+                self.cpp_info.components[alias].requires.append("qt")
+            
+            if name == "VTK::RenderingOpenGL2":
+                self.cpp_info.components[alias].requires.append("opengl/system")
```

Comment:
@prince-chrismc , I found some time this week to work on this. I feel I am getting close. But the CI throws an error about not finding opengl. I added this line and opengl does get added to the list of requires. But conan still throws and error about package require 'opengl' not used in components requires. Did I do it right?